### ISSUE TITLE
[Feat] CI: add a link (and email) linter

### ISSUE THREAD

**Issue by neteler** [OSGeo/grass#3198]
Add the lychee linter:

https://github.com/lycheeverse/lychee

It is a link (and email) linter, that checks if all the links are valid and alive. It will flag links that 404 for example. It was a relatively new addition to Megalinter and it proved itself a couple times in the last couple months.

_Originally posted by @echoix in https://github.com/OSGeo/grass/issues/3157#issuecomment-1728554395_

**Comment by neteler** [comment]
@echoix would this still be of interest?

**Comment by echoix** [comment]
Yes, kinda. It got rediscussed not too long ago, a couple months.
